### ISSUE TITLE
create a document from a stream

### ISSUE THREAD

**Issue by Mouvedia** [whatwg/streams#520]
cf https://github.com/whatwg/fetch/issues/16#issuecomment-143769167 by @annevk

references:
whatwg/fetch#151
whatwg/fetch#16
bitinn/node-fetch#113

**Comment by domenic** [comment]
No, this is out of scope for the streams specification, sorry you never got a response. The streams specification defines streams; it does not deal with how to create them from specific web platform objects.

**Comment by Mouvedia** [comment]
@domenic could you tell me **on which repo this request belong**?

The `timeout` property and the document response type are missing from `fetch` to be on par with `XMLHttpRequest`. `timeout` is covered by whatwg/fetch#179.

**Comment by annevk** [comment]
@Mouvedia not sure if @domenic agrees, but I guess arguably you could request a streaming parser for HTML in whatwg/html... But taking a step back, this idea is known, but what needs to happen is a whole bunch of work to figure things out.

**Comment by Mouvedia** [comment]
@annevk I have referenced 5 different issues on 4 repositories*; the community has spoken, something is missing. Could you fill the issue yourself? In layman's terms we need a builtin `document` or `xml` method on `Response`. I am sure you will be able to reformulate that in a sensible way.

* https://github.com/whatwg/streams/issues/520#issue-178056650

**Comment by annevk** [comment]
`Response` won't get such a method. If that is all you need, use `DOMParser` on `response.text()`.

**Comment by Mouvedia** [comment]
@annevk that means that users will have to handle response of type document in a separate path or extend Response with a custom method of their own. Is that really the best we can do? Even if json has clearly won the battle over xml in terms of usage, it doesn't mean that a dedicated method is not necessary. The parity with XHR should be of utmost importance IMHO. The overall experience shouldn't involve hacks: it should be consistent.

**Comment by annevk** [comment]
I guess we disagree on that.

**Comment by Mouvedia** [comment]
It's really hard to believe you. Who would want fetch to be lacking stuff that XHR could do?
Parity is one of the most essential goal for `fetch()` to be considered a serious alternative for **all** use cases.
